### ISSUE TITLE
Allow `exclude` to work with DjangoModelFactory :: create objects without an `id` attribute

### ISSUE THREAD

**Issue by hannylicious** [FactoryBoy/factory_boy#1003]
#### The problem

I have a use case where I need to create a Django model that _does not include an id_ ; 

I haven't found any good way to do this - as using the `exclude` does not appear to work with Django model factories (I have tried `exclude=('id', )` on the factory where I need this functionality).

The model I need to do this on is an unmanged django model, so there are no migrations for this model and this model does _not_ have any primary key (see 'extra notes' as to why).

#### Proposed solution

Give a way for the `exclude` functionality to work on `DjangoModelFactory` objects _and_ allow it to work in excluding `id` attributes.

#### Extra notes

I would just create it as a regular `Factory`, but I need to be able to save the data to a specific database and I use the `django_get_or_create` functionality in my data creation scripts to tie all the data together.

I realize this is a _very particular and specific_ use case. The reason behind it is this: in the production environment we have some pretty robust SQL Views that pull data from numerous tables. Rather than having to create all of those tables filled with data and then recreate the sql view locally in Docker - we create a `DjangoModelFactory` and use that to 'mock' the data we need in a table. But by default the factory creates an `id` on that model, which causes issues.

If this kind of functionality exists already? I couldn't find it in the documentation. 

Any thoughts/ideas as to how to accomplish this are appreciated!

**Comment by rbarrois** [comment]
Hi,

Thanks for the report!

I'm not sure I understand your issue accurately; `factory_boy` doesn't force an ID.

The underlying steps are:
1. Combine the class-level declarations with the overrides provided when calling the factory;
2. Evaluate all declarations, building a dictionary of "resolved" declarations;
3. Apply all `exclude=`, `rename`, etc. on that dictionary;
4. Pass the dictionary as kwargs to either `MyModel(**kwargs)` (for `Factory.build()`) or `MyModel.objects.create(**kwargs)` (for `Factory.create()` or `Factory()`)

If you're not providing a value for `id` in your factory declaration or when calling the factory in your test, there won't be any value passed to the underlying model' `__init__` or `create`.

Can you provide a minimal example of the model and factory you're using?

If possible, I would also be interested in having an example of "pure Django" code that works as you expect it, i.e code written without using any factory:

```python
# Maybe something along those lines?
instance, created = MyModel.objects.get_or_create(x=1, y=2, defaults=dict(z=3, a=42))
```

**Comment by hannylicious** [comment]
Hi, thanks for the response!

In setting up an example, I came across a realization that reinforces what you said above. 

In our models - because they're unmanaged and have no migrations in order to set them up locally we allow Django to 'manage' the models locally in tests like this:

```
class Thing(models.Model):
    person_id = models.IntegerField(db_column="personID")
    state_id = models.IntegerField(db_column="stateID")
    activity_id = models.IntegerField(db_column="activityID")
    first_name = models.CharField(db_column="firstName", max_length=35)
    last_name = models.CharField(db_column="lastName", max_length=40)

    class Meta(object):
        managed = getattr(settings, "UNDER_TEST", False)
        db_table = "[dbo].[v_thing]"
```

So when the variable "UNDER_TEST" is set (which we set during testing) the model is considered `managed` by Django. 

That is what is causing the `id` attribute to be created automatically - because a Django managed model requires a primary key of some form to be placed on the model. So the issue I was facing is because of Django.

FactoryBoy is working exactly as intended! 

Because of that, I will be closing this request. Thanks for all the work and for such a great tool!
